components/AudioPlayer.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useRef, useEffect } from 'react'
+import styles from "../components/AudioPlayer.module.css";
+import utilsStyles from '../styles/utils.module.css';
+import Image from 'next/image';
+import { BsArrowLeftShort } from "react-icons/bs"
+import { BsArrowRightShort } from "react-icons/bs"
+import { FaPlay } from "react-icons/fa"
+import { FaPause } from "react-icons/fa"
+//
+
+const AudioPlayer = () => {
+  // state
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [duration, setDuration] = useState(0);
+  const [currentTime, setCurrentTime] = useState(0);
+
+  // references
+  const audioPlayer = useRef();   // reference our audio component
+  const progressBar = useRef();   // reference our progress bar
+  const animationRef = useRef();  // reference the animation
+
+  useEffect(() => {
+    const seconds = Math.floor(audioPlayer.current.duration);
+    setDuration(seconds);
+    progressBar.current.max = seconds;
+  }, [audioPlayer?.current?.loadedmetadata, audioPlayer?.current?.readyState]);
+
+  const calculateTime = (secs) => {
+    const minutes = Math.floor(secs / 60);
+    const returnedMinutes = minutes < 10 ? `0${minutes}` : `${minutes}`;
+    const seconds = Math.floor(secs % 60);
+    const returnedSeconds = seconds < 10 ? `0${seconds}` : `${seconds}`;
+    return `${returnedMinutes}:${returnedSeconds}`;
+  }
+
+  const togglePlayPause = () => {
+    const prevValue = isPlaying;
+    setIsPlaying(!prevValue);
+    if (!prevValue) {
+      audioPlayer.current.play();
+      animationRef.current = requestAnimationFrame(whilePlaying)
+    } else {
+      audioPlayer.current.pause();
+      cancelAnimationFrame(animationRef.current);
+    }
+  }
```

Comment:
```suggestion
  const togglePlayPause = () => {
    setIsPlaying((prevState) => !prevState);
    if (!isPlaying) {
      audioPlayer.current.play();
      animationRef.current = requestAnimationFrame(whilePlaying)
    } else {
      audioPlayer.current.pause();
      cancelAnimationFrame(animationRef.current);
    }
  }
```
Maybe this is a personal preference, but I would write this way. (**no need to change it tho**)